dropwizard-db/src/main/java/io/dropwizard/db/TimeBoundHealthChecks.java

```diff
@@ -0,0 +1,19 @@
+package io.dropwizard.db;
+
+import com.codahale.metrics.health.HealthCheck;
+import io.dropwizard.util.Duration;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+
+public class TimeBoundHealthChecks {
+    public static HealthCheck.Result check(ExecutorService executorService, Duration timeBound, Callable<HealthCheck.Result> c) {
```

Comment:
I think it's better to make `ExecutorService` and `Duration` as class dependencies rather than method arguments.
Doing so, we can encapsulate them in `TimeBoundHealthChecks` and then inject it to `DBIHealthCheck` and `SessionFactoryHealthCheck`. As the result we have a parametrized service that accepts Callables, rather than
a utility method. 

Something like this:
````java
public class TimeBoundHealthChecks {

    private final ExecutorService executorService;
    private final Duration timeBound;

    public TimeBoundHealthChecks(ExecutorService executorService, Duration timeBound) {
        this.executorService = executorService;
        this.timeBound = timeBound;
    }

    public HealthCheck.Result check(Callable<HealthCheck.Result> callable) {
        try {
            return executorService.submit(callable).get(timeBound.getQuantity(), timeBound.getUnit());
        } catch (Exception e) {
            return HealthCheck.Result.unhealthy(Throwables.getRootCause(e));
        } 
    }
}
````

````java
public class DBIHealthCheck extends HealthCheck {
    private final DBI dbi;
    private final String validationQuery;
    private final TimeBoundHealthChecks timeBoundHealthChecks;

    public DBIHealthCheck(ExecutorService executorService, Duration duration, DBI dbi, String validationQuery) {
        this.dbi = dbi;
        this.validationQuery = validationQuery;
        timeBoundHealthChecks = new TimeBoundHealthChecks(executorService, duration);
    }

    public DBIHealthCheck(DBI dbi, String validationQuery) {
        this(MoreExecutors.newDirectExecutorService(), Duration.seconds(0), dbi, validationQuery);
    }

    @Override
    protected Result check() throws Exception {
        return timeBoundHealthChecks.check(new Callable<Result>() {
            @Override
            public Result call() throws Exception {
                try (Handle handle = dbi.open()) {
                    handle.execute(validationQuery);
                    return Result.healthy();
                }
            }
        });
    }

}
````